### PR TITLE
ENH track the masked fraction

### LINKED ISSUE #48: Count fraction of input images masked at each pixel rather than rely on ormask
This may explain why cutting on masked fraction wasn't helping as much as we expected

### PR DESCRIPTION
**Pull request by beckermr**
This PR explicitly tracks the masked fraction.

closes #48

### CODE REVIEW

**Comment by beckermr** [comment]
ok @esheldon This one is ready for review!

File: descwl_coadd/coadd_old.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def _process_exps(self):
                     if self.interp_bright:
                         flag_bright_as_sat_dm(mask=bmask)
 
-                    iimage, inoise = interpolate_image_and_noise(
```
Comment (esheldon) [inline descwl_coadd/coadd_old.py:192]: I think we can delete this whole module

File: descwl_coadd/coadd.py [hunk 2]
```diff
@@ -802,13 +861,15 @@ def _finish_init(self):
             jacobian=jac,
             psf=psf_obs,
             store_pixels=False,
+            mfrac=mfrac,
         )
 
         flags_for_maskfrac = self.coadd_exp.mask.getPlaneBitMask('BRIGHT')
-        self.meta['mask_frac'] = get_masked_frac(
+        self.meta['bright_frac'] = get_masked_frac(
```
Comment (esheldon) [inline descwl_coadd/coadd.py:868]: need to make sure this rename is ok downstream

File: descwl_coadd/coadd.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def make_coadd_obs(
     CoaddObs (inherits from ngmix.Observation)
     """
 
-    coadd_exp, coadd_noise_exp, coadd_psf_exp = make_coadd(
+    coadd_exp, coadd_noise_exp, coadd_psf_exp, coadd_mfrac_exp = make_coadd(
```
Comment (esheldon) [inline descwl_coadd/coadd.py:82]: this return list is getting long.  should we convert to dict to avoid bugs?

**Review by esheldon — COMMENTED** [review]
looks good, minor comments

File: descwl_coadd/coadd.py (hunk 2 above)
Comment (beckermr) [reply descwl_coadd/coadd.py:868]: Do you want to run a quick test or maybe just change the downstream code? If you point to where the change needs to go, I can make it.

File: descwl_coadd/coadd.py (hunk 3 above)
Comment (beckermr) [reply descwl_coadd/coadd.py:82]: That's a good idea. People also like named tuples for this kind of thing, but I find dicts more intuitive.

File: descwl_coadd/coadd.py (hunk 2 above)
Comment (esheldon) [reply descwl_coadd/coadd.py:868]: you are also saving bright_frac so I can adjust downstream if needed

File: descwl_coadd/coadd_old.py (hunk 1 above)
Comment (beckermr) [reply descwl_coadd/coadd_old.py:192]: done.

File: descwl_coadd/coadd.py (hunk 3 above)
Comment (beckermr) [reply descwl_coadd/coadd.py:82]: done.

**Comment by beckermr** [comment]
This is ready for another look @esheldon!

File: descwl_coadd/tests/test_coadd.py [hunk 4]
```diff
@@ -98,6 +100,56 @@ def test_coadds_smoke(dither, rotate):
...
+
+        coadd_dims = (coadd_dim, )*2
+        psf_dims = (psf_dim, )*2
+        assert coadd.image.shape == coadd_dims
+        assert coadd.psf.image.shape == psf_dims
+
+        assert np.all(np.isfinite(coadd.psf.image))
+        assert not np.all(coadd.mfrac == 0)
+        assert np.max(coadd.mfrac) > 0.1
+        assert np.mean(coadd.mfrac) < 0.05
```
Comment (beckermr) [inline descwl_coadd/tests/test_coadd.py:145]:
```suggestion
        assert np.mean(coadd.mfrac) < 0.05
        assert np.all(coadd.mfrac >= 0)
        assert np.all(coadd.mfrac <= 1)
```

**Review by esheldon — APPROVED** [review]

**Pull request merged**